downloader.py

```diff
@@ -111,9 +121,49 @@ def grb2_to_array(filename):
     for i, level in enumerate(levels):
         assert(u[i]['level'] == level)
         assert(v[i]['level'] == level)
-        dataset[0][i] = u[i].data()[0][::2, ::2]
+        dataset[0][i] = u[i].data()[0][::2, ::2] # Takes the second element of every second array in data
         dataset[1][i] = v[i].data()[0][::2, ::2]
     return dataset
 
+## save data as npz file of ['data', 'timestamp (unix)', 'interval', 'levels']
+def combine_files():
+    file_list = os.listdir(f'{args.savedir}/temp')
+    filesets = [['']]
+    
+    for i in range(NUM_MEMBERS-1):
+        filesets.append([''])
+    for f in file_list:
+        n = int(f[26:28])
+        filesets[n-1].append(f)
+
+    for i in filesets:
+        member = i[1][26:28]
+        data = combine_npy_for_member(member , i[1:])
+        data = np.float16(data)
+        logger.info(f'Completed combining files for {member}')
+        
+        savename = str(i[1][0:11] + str(member).zfill(2) + ".npz")
+        dt = datetime(int(savename[0:4]), int(savename[4:6]), int(savename[6:8]), int(savename[8:10]))
```

Comment:
For maintainability, please do not form a string only to split it apart in a potentially confusing way. If you want to access a datetime object, use datetime.strptime. Additionally, since you just saved the files, you already know the timestamp so you don't have to reconstruct it from the file names.